internal/helpers/rollout_restart.go

```diff
@@ -140,6 +151,10 @@ func patchForRolloutRestart(ctx context.Context, obj ctrlclient.Object, client c
 		}
 		t.Spec.Template.ObjectMeta.Annotations[AnnotationRestartedAt] = time.Now().Format(time.RFC3339)
 		return client.Patch(ctx, t, patch)
+	case *argorolloutsv1alpha1.Rollout:
+		patch := ctrlclient.MergeFrom(t.DeepCopy())
```

Comment:
I'm curious why this uses `MergeFrom()` instead of `StrategicMergeFrom()` like the other cases?

Reply:
`StrategicMergeFrom()` doesn't support CRDs. I've added a comment for the explaination https://github.com/hashicorp/vault-secrets-operator/blob/461cf779fab39327ab4a1460236b63a3b70bfff4/internal/helpers/rollout_restart.go#L163